File: model/lead.py
"""Lead ORM model."""
from __future__ import annotations
from typing import Optional

from sqlalchemy import Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from . import Base


class Lead(Base):
    __tablename__ = "leads"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    name: Mapped[str] = mapped_column(String(255), nullable=False)
    email: Mapped[Optional[str]] = mapped_column(String(255), index=True)
    phone: Mapped[str] = mapped_column(String(30), nullable=False, index=True)
    subject: Mapped[Optional[str]] = mapped_column(String(255))
    message: Mapped[Optional[str]] = mapped_column(Text)

File: model/lead.js
export class Lead {
  static tableName = 'leads';
  static schema = {
    id: { type: 'integer', primaryKey: true, autoIncrement: true },
    name: { type: 'string', maxLength: 255, required: true },
    email: { type: 'string', maxLength: 255, index: true },
    phone: { type: 'string', maxLength: 30, required: true, index: true },
    subject: { type: 'string', maxLength: 255 },
    message: { type: 'text' },
  };

  constructor(attrs = {}) {
    Object.assign(this, attrs);
  }
}
